### PR TITLE
Validate release charts

### PR DESCRIPTION
**Pull request by nicholasSUSE**
### Issue: https://github.com/rancher/ecm-distro-tools/issues/469

This solution is only for `Checkpoint: 0`, defined in the abovementioned issue.  

### Checkpoint 0: Validate **release.yaml**

Validation steps:
- [x] Each chart version in **release.yaml** DOES NOT modify an already released chart. If so, stop and modify the versions so that it releases a net-new chart.
- [x] Each chart version in **release.yaml** IS exactly 1 more patch/minor version than the last released chart version. If not, stop and modify the versions so that it releases a net-new chart.

### CODE REVIEW

**Comment by nicholasSUSE** [comment]
@rancher/release-team
I know the linter is failing, I am on to fix the errors.

**Comment by nicholasSUSE** [comment]
Ok, I started working on the code the linter pointed, and unit tests started breaking. 
I will separate some time to refactor this, for now disregard it.

**Review by tashima42 — CHANGES_REQUESTED** [review]

File: main.go [hunk 1]
```diff
@@ -714,3 +764,31 @@ func release(c *cli.Context) {
...
+	if Branch == "" {
+		logrus.Fatal("BRANCH environment variable must be set to run validate-release-charts")
+	}
+
+	r := filesystem.GetFilesystem(getRepoRoot())
+
+	if !strings.HasPrefix(Branch, "release-v") {
+		fmt.Println("Branch must be in the format release-v2.x")
+		os.Exit(1)
+	}
```
[the remark this answers is not in the document]
Comment (nicholasSUSE) [reply main.go:784]: I will do that in time. 
For now, I am just updating whatever I am working on.

[the remark this answers is not in the document]
Comment (tashima42) [reply main.go:784]: It's fine not changing other functions now, but I really think in the same function, the same type of fatal should be used. e.g:
```diff
- logrus.Fatalf("encountered error while initializing d: %v", err)
+ fmt.Printf("encountered error while initializing d: %v", err)
+ os.Exit(1)
```

[the remark this answers is not in the document]
Comment (nicholasSUSE) [reply main.go:787]: @tashima42 now ready

**Comment by nicholasSUSE** [comment]
Bypassing linter until refactoring of legacy code

**Pull request merged**